### DISCUSSION TITLE
How to use idl file to send a socket msg

### DISCUSSION THREAD

**Discussion by Crusoekid** [eProsima/Fast-DDS#4113] [Support]
when i see the example `TestWriterSocket.cpp` ,it only show me how to send a int value to Reader;how can i send a image to reader by idl. THX

**Comment by JesusPoderoso** [comment]
Hi @Crusoekid, thanks for usinig Fast DDS.
The `run` method of the [TestWriterSocket.cpp example](https://github.com/eProsima/Fast-DDS/blob/7c75b6d5294e31781af17238461d10c95c3d92ff/examples/cpp/rtps/AsSocket/TestWriterSocket.cpp#L90)  loads the sample into the `CacheChange_t`, which goes directly to the writer history. You should modify the this section of the code where the CacheChange_t' serializedPayload data is set:
```
#if defined(_WIN32)
        ch->serializedPayload.length =
                sprintf_s((char*)ch->serializedPayload.data, 255, "My example string %d", i) + 1;
#else
        ch->serializedPayload.length =
                snprintf((char*)ch->serializedPayload.data, 255, "My example string %d", i) + 1;
#endif // if defined(_WIN32)
```
So instead of setting the data to be sent as a `bounded string<255>` that contains the `int` value, you add your image.
You can do it, for instance, by introducing your images in the data buffer as an `octet sequence`.
